client/src/components/student/StudentView.jsx

```diff
@@ -56,11 +68,31 @@ const StudentView = () => {
           addNewCourse={handleCourseAddition}
         />
       </Modal>
+
       <div className="grid grid-cols-4 grid-flow-row gap-3">
         {courses.map((course, idx) => (
+          <div key={idx} onClick={() => handleCourseClick(course)}>
           <StudentCourseDetails key={idx} course={course} />
+          {/*button to view teammates should display only when a course gets selected 
+          check with the course id since it's unique*/}
+          {/*course exists and is the one selected */}
+          {selectedCourse && selectedCourse.id === course.id && (
+            <button
+              className="text-sm mt-2 p-1 border rounded-md bg-blue-500 text-white"
+              onClick={handleOpenTeam} /*Not sure if this button was already implemented or not, but
+               here I'm adding it just in case: when you click it it renders the next Modal*/
+            >
+              View my Team
+            </button>
+          )}
+        </div>
         ))}
       </div>
+      {/*Displays the Team view modal */}
+      <Modal open={isViewingTeam} onClose={handleCloseTeam}>
```

Comment:
When a course gets clicked it should render the interface from MyTeam.jsx. I understood that I had to implement it as a modal instead of creating a new path for it